internal/provider/context_field_resource.go

```diff
@@ -0,0 +1,320 @@
+}
+
+func (r *contextFieldResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	tflog.Debug(ctx, "Preparing to read contextField resource")
+	var state contextFieldResourceModel
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	contextField, httpRes, err := r.client.ContextAPI.GetContextField(context.Background(), state.Name.ValueString()).Execute()
+	if !ValidateApiResponse(httpRes, 200, &resp.Diagnostics, err) {
+		return
+	}
+
+	state.hydrateFromApi(*contextField)
+	resp.State.Set(ctx, &state)
+
+	tflog.Debug(ctx, "Finished reading contextField resource", map[string]interface{}{"success": true})
+}
+
+func (r *contextFieldResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	tflog.Debug(ctx, "Preparing to update contextField resource")
+	var plan contextFieldResourceModel
+	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	updateContextFieldRequest := *unleash.NewUpdateContextFieldSchemaWithDefaults()
+
+	var contextErr = populateContextField(&updateContextFieldRequest, plan)
+	if contextErr != nil {
+		resp.Diagnostics.AddError("error populating context field", contextErr.Error())
+		return
+	}
+
+	var httpRes, err = r.client.ContextAPI.UpdateContextField(context.Background(), plan.Name.ValueString()).UpdateContextFieldSchema(updateContextFieldRequest).Execute()
+	if !ValidateApiResponse(httpRes, 200, &resp.Diagnostics, err) {
+		return
+	}
+
+	contextField, httpRes, err := r.client.ContextAPI.GetContextField(context.Background(), plan.Name.ValueString()).Execute()
+	if !ValidateApiResponse(httpRes, 200, &resp.Diagnostics, err) {
+		return
+	}
+
+	plan.hydrateFromApi(*contextField)
+	resp.State.Set(ctx, &plan)
+
+}
+
+func (r *contextFieldResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	tflog.Debug(ctx, "Preparing to delete contextField resource")
+	var state contextFieldResourceModel
+	diags := req.State.Get(ctx, &state)
+	resp.Diagnostics.Append(diags...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	httpRes, err := r.client.ContextAPI.DeleteContextField(context.Background(), state.Name.ValueString()).Execute()
+	if !ValidateApiResponse(httpRes, 200, &resp.Diagnostics, err) {
+		return
+	}
+
+	resp.State.RemoveResource(ctx)
+	tflog.Debug(ctx, "Finished deleting contextField resource", map[string]interface{}{"success": true})
+}
+
+func (m *contextFieldResourceModel) hydrateFromApi(api unleash.ContextFieldSchema) {
+	m.Name = types.StringValue(api.Name)
+	if api.Description.IsSet() && api.Description.Get() != nil {
+		m.Description = types.StringValue(*api.Description.Get())
+	} else {
+		m.Description = types.StringNull()
+	}
+
+	if api.Stickiness != nil {
+		m.Stickiness = types.BoolValue(*api.Stickiness)
+	} else {
+		m.Stickiness = types.BoolNull()
```

Comment:
Doesn't look like this can ever happen but I've chosen to be honest about the result if it does